Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,342 @@
+---
+title: Implicitly-scoped qubit allocation
+description: Provides more convenient syntax for allocating qubits.
+author: Sarah Marshall
+date: September 28, 2020
+---
+
+# Proposal
+
+1. Add new `use` and `borrow` statements to allocate qubits that are valid until the end of the current block.
+2. Add new scope statement that does not require a preceding keyword, and is executed exactly once; equivalent to `if (true) { }`.
+3. Deprecate and remove existing `using` and `borrowing` block statements.
+
+# Justification
+
+Qubit allocation is currently awkward in Q# for two reasons:
+
+1. It requires creating a new block to define the lifetime of the qubit, even when the intended lifetime is the same as the current block.
+2. Allocating multiple qubits at the same time requires either tuple destructuring or nested blocks, both of which can be hard to read.
```

Comment:
Absolutely agreed, especially because it often feels like qubit allocation in `using` blocks is "transposed" for more complicated allocations:
```
using ((a, b, c) = (Qubit(), Qubit(), Qubit())) {
    // ...
}
```
Never mind that requiring `)))` for a relatively common case feels a bit heavy-weight.